scipy/optimize/tests/test_zeros.py

```diff
@@ -619,3 +619,22 @@ def fpp(x):
     # Check that it now succeeds.
     rt, r = newton(f, x0, fprime=fp, fprime2=fpp, full_output=True)
     assert(r.converged)
+
+
+def test_gh_9608_preserve_array_shape():
+    """
+    Test that shape is preserved for array inputs even if fprime or fprime2 is
+    scalar
+    """
+    def f(x):
+        return x**2
+
+    def fp(x):
+        return 2 * x
+
+    def fpp(x):
+        return 2
+
+    x0 = np.array([-2], dtype=np.float32)
+    rt, r = newton(f, x0, fprime=fp, fprime2=fpp, full_output=True)
+    assert(r.converged)
```

Comment:
you need to use `np.testing.assert_` instead of a bare assert

Reply:
I don't think that's the case now that we use pytest

Reply:
(see #8583)

Reply:
Is there a test that newton fails for a mismatch between f and app when passed a longer array?  E.g. continuing the example above, something like 
```
x0 = np.array([-2, -3], dtype=np.float32)
# This next invocation should fail
rt, r = newton(f, x0, fprime=fp, fprime2=fpp, full_output=True)
```

Reply:
I looked through `test_zeros.py` but there is no such test currently. Should I add a test to catch the exception?

Reply:
It's now part of the contract in the docstring so adding a check on a mismatch would be reasonable.